addons/project/models/project.py

```diff
@@ -1475,16 +1473,11 @@ def _compute_dependent_tasks_count(self):
             for task in tasks_with_dependency:
                 task.dependent_tasks_count = dependent_tasks_count_dict.get(task.id, 0)
 
-    @api.depends('partner_id.phone')
-    def _compute_partner_phone(self):
-        for task in self:
-            if task.partner_phone != task.partner_id.phone:
-                task.partner_phone = task.partner_id.phone
-
-    def _inverse_partner_phone(self):
-        for task in self:
-            if task.partner_id and task.partner_phone != task.partner_id.phone:
-                task.partner_id.phone = task.partner_phone
+    @api.depends('project_id')
+    def _compute_should_display_partner(self):
+        non_private_tasks = self.filtered('project_id')
+        non_private_tasks.should_display_partner = True
+        (self - non_private_tasks).should_display_partner = False
```

Comment:
Is it really needed why not using `project_id` or even `is_private` field in the attrs instead? 🤔

Reply:
I added a field to avoid so many `xpath` to display partner for `allow_billable` and `is_fsm` task. If not this then I'll have to inherit all views in `sale_project` to hide partner for non billable and inherit all view in `industry_fsm_sale` to hide partner for non billable/non fsm project and task. Is it ok? :thinking:

Reply:
I think it is okay to update the domains instead of creating a field for that. 🙂 

Normally, those domains don't change many times so, it is acceptable. 🙂
I hope I answered to all your questions about that. If not, then don't hesitate to ping me in direct message. 🙂